MetaMask/metamask-extension
# Fixed a duplicate account import test case relying on an upstream typo

## Explanation



This pull request in the upstream KeyringController https://github.com/MetaMask/KeyringController/pull/153 fixes the issue https://github.com/MetaMask/metamask-extension/issues/16370.

This current pull request fixes a test case such that the upstream KeyringController fix can be used.

Currently, users encounter a typo when attempting to import a duplicate account into MetaMask. The typo is not localized and originates from the upstream KeyringController. I have submitted a pull request here https://github.com/MetaMask/KeyringController/pull/153 to fix the typo, but doing so would break a test case in this repository which expects the typo. This pull request fixes the test case such that it will expect the typo-free error from KeyringController. This pull request should be merged after https://github.com/MetaMask/KeyringController/pull/153 because it is dependent on the typo being fixed there.

## More Information



I encountered this typo when attempting to import an account via its private key after that account had previously already been imported.

## Screenshots/Screencaps



### Before



![before](https://user-images.githubusercontent.com/2444641/199836898-1b0f6b17-91e6-4cf8-9df4-0ff96535eca6.png)

### After



The UI will look the same until after https://github.com/MetaMask/KeyringController/pull/153 is merged and incorporated. After that point, the end user will no longer see the "you're are" typo and will just see "you are".

## Manual Testing Steps



You can verify that this issue is present by importing an account from private key and then attempting to import the same account. You should no longer see the typo if using the new KeyringController.

## Pre-Merge Checklist


### + If there are functional changes:
